python/ground_git.py

```diff
@@ -0,0 +1,1056 @@
+                          fromNodeVersionEndId=None, reference=None, referenceParameters=None, tags=None,
+                          structureVersionId=None, parentIds=None):
+        self._check_init()
+        body = self._get_rich_version_json(EdgeVersion.__name__, reference, referenceParameters,
+                                           tags, structureVersionId, parentIds)
+
+        body["edgeId"] = edgeId
+        body["toNodeVersionStartId"] = toNodeVersionStartId
+        body["fromNodeVersionStartId"] = fromNodeVersionStartId
+
+        if toNodeVersionEndId > 0:
+            body["toNodeVersionEndId"] = toNodeVersionEndId
+
+        if fromNodeVersionEndId > 0:
+            body["fromNodeVersionEndId"] = fromNodeVersionEndId
+
+        edgeVersion = EdgeVersion(body)
+        edgeVersionId = edgeVersion.get_id()
+
+        #self.edgeVersions[edgeVersionId] = edgeVersion
+
+        write = self._deconstruct_rich_version_json(body)
+        self._write_files(edgeVersionId, write)
+        self._commit(edgeVersionId, EdgeVersion.__name__)
+
+        return edgeVersionId
+
+    def getEdge(self, sourceKey):
+        self._check_init()
+        return self._read_files(sourceKey, Edge.__name__)
+
+
+    def getEdgeLatestVersions(self, sourceKey):
+        self._check_init()
+        edgeVersionMap = self._read_all_version(sourceKey, EdgeVersion.__name__, Edge.__name__)
+        edgeVersions = set(list(edgeVersionMap.keys()))
+        is_parent = set([])
+        for evId in edgeVersions:
+            ev = edgeVersionMap[evId]
+            if ('parentIds' in ev) and (ev['parentIds']):
+                assert type(ev['parentIds']) == list
+                for parentId in ev['parentIds']:
+                    is_parent |= {parentId, }
+        return [edgeVersionMap[Id] for Id in list(edgeVersions - is_parent)]
+
+    def getEdgeHistory(self, sourceKey):
+        self._check_init()
+        edgeVersionMap = self._read_all_version(sourceKey, EdgeVersion.__name__, Edge.__name__)
+        edgeVersions = set(list(edgeVersionMap.keys()))
+        parentChild = {}
+        for evId in edgeVersions:
+            ev = edgeVersionMap[evId]
+            if ('parentIds' in ev) and (ev['parentIds']):
+                assert type(ev['parentIds']) == list
+                for parentId in ev['parentIds']:
+                    if not parentChild:
+                        edgeId = ev['edgeId']
+                        parentChild[str(edgeId)] = parentId
+                    parentChild[str(parentId)] = ev['id']
+        return parentChild
+
+    def getEdgeVersion(self, edgeVersionId):
+        self._check_init()
+        return self._read_version(edgeVersionId, EdgeVersion.__name__)
+
+    ### NODES ###
+    def createNode(self, sourceKey, name="null", tags=None):
+        self._check_init()
+        if not self._find_file(sourceKey, Node.__name__):
+            body = self._create_item(Node.__name__, sourceKey, name, tags)
```

Comment:
We seem to be reusing the class names a lot here. Not necessarily a bad thing, but can we think of a more efficient solution?